storages/customer_storage.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from models.customer import Customer
+from config.db_conn import db_conn
+from typing import List
+
+class CustomerStorage:
+    def __init__(self):
+        self.logger = logging.getLogger(__name__)
+        self.db = db_conn
+    
+    def get_customer_by_id(self, id: str) -> Customer:
+        self.logger.info("Getting an customer in DB")
+        try:
+            with self.db.cursor() as cursor:
+                cursor.execute(
+                    """
+                    SELECT id, name, email, created_at, updated_at
+                    FROM customers
+                    WHERE id = %s;
+                    """,
+                    (id,))
+                
+                result = cursor.fetchone()
+
+                if result == None:
```

Comment:
Adicionando esse raise aqui, o tratamento que vc fez la no router nunca vai acontecer. Pq vc nunca vai retornar vazio, sempre ta lançando uma exception.  
Recomendo aqui lançar uma exceção mais específica, capturar ela no router e retornar oq vc quer (404) com base nisso.

Reply:
Sobre exceção especifica, vc pode usar o ValueError,  ela é usada pra representar situações em que um valor esperado não é encontrado ou é inválido. Embora não seja especificamente para "not found", ela é adequada para indicar que o valor do ID fornecido não corresponde a um cliente existente. 

Ou pode criar usa propria exception como um CustomerNotFoundException.

Reply:
Resolvi no commit [f5ad3a0](https://github.com/michaelrodriguess/customer-service/pull/9/commits/f5ad3a02ab25abb266d33d15a163c826abb3fa44)